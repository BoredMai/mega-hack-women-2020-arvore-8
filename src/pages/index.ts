import { Page } from 'types';

export const skillPage: Page = {
  content: `A princesa {nome} era muito conhecida por ser uma exímia...`,
};

export const rootPage: Page = {
  content: `Era uma vez um reino misterioso e sagrado onde habitavam seres mágicos, um povoado e sua princesa! Como ela era amada pelo seu povo! Sua inteligência, coragem e lealdade para com seu povo era admirável.\nSeu nome era...`,
  next: [skillPage],
  prompt: {
    key: 'characterName',
    type: 'text',
    placeholder: 'Nome da princesa',
  },
};
