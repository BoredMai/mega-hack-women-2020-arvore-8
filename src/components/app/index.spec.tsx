import { shallow, ShallowWrapper } from 'enzyme';
import React from 'react';

import { rootPage } from 'pages';
import App from '.';

describe('<App>', () => {
  const setPageMock = jest.fn().mockName('setPage');
  const setStoryPromptMock = jest.fn().mockName('setStoryPrompt');
  const useStateSpy = jest.spyOn(React, 'useState');
  let wrapper: ShallowWrapper;

  beforeEach(() => {
    jest.clearAllMocks();
    useStateSpy
      .mockImplementationOnce(() => [rootPage, setPageMock])
      .mockImplementationOnce(() => [{}, setStoryPromptMock]);
    wrapper = shallow(<App />) as ShallowWrapper;
  });

  it('renders properly', () => {
    expect(wrapper.getElement()).toMatchSnapshot();
  });

  it('updates state correctly on promptUpdate', () => {
    const promptKey = 'promptKey';
    const promptValue = 'promptValue';
    wrapper
      .find('Storyteller')
      .simulate('promptUpdate', promptKey, promptValue);

    expect(setStoryPromptMock).toHaveBeenCalledWith({ promptKey: promptValue });
    expect(setPageMock).toHaveBeenCalledWith(rootPage.next[0]);
  });
});
