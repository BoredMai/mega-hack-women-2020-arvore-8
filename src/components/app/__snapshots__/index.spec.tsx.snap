// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<App> renders properly 1`] = `
<div
  className="main"
>
  <Header />
  <Storyteller
    onPromptUpdate={[Function]}
    page={
      Object {
        "content": "Era uma vez um reino misterioso e sagrado onde habitavam seres mágicos, um povoado e sua princesa! Como ela era amada pelo seu povo! Sua inteligência, coragem e lealdade para com seu povo era admirável.
Seu nome era...",
        "next": Array [
          Object {
            "content": "A princesa {nome} era muito conhecida por ser uma exímia...",
          },
        ],
        "prompt": Object {
          "key": "characterName",
          "placeholder": "Nome da princesa",
          "type": "text",
        },
      }
    }
  />
</div>
`;
